model_analyzer/config/generate/manual_model_config_generator.py

```diff
@@ -37,24 +39,96 @@ def __init__(self, config, model, client, variant_name_manager,
         default_only: Bool 
             If true, only the default config will be generated
             If false, the default config will NOT be generated                
+        early_exit_enable: Bool
+            If true, the generator can early exit if throughput plateaus
         """
         super().__init__(config, model, client, variant_name_manager,
-                         default_only)
+                         default_only, early_exit_enable)
 
         self._reload_model_disable = config.reload_model_disable
         self._num_retries = config.client_max_retries
         self._search_disabled = config.run_config_search_disable
         self._curr_config_index = 0
+        self._curr_max_batch_size_index = 0
+        self._max_batch_size_warning_printed = False
+
+        self._max_batch_sizes = None
+        self._non_max_batch_size_param_combos = []
+        self._determine_max_batch_sizes_and_param_combos()
+
         self._configs = self._generate_model_configs()
 
+        # Contains the max throughput from each provided list of measurements
+        # since the last time we stepped max_batch_size
+        #
+        self._curr_max_batch_size_throughputs = []
+
     def _done_walking(self):
+        return self._done_walking_configs() \
+           and self._done_walking_max_batch_size()
+
+    def _done_walking_configs(self):
         return len(self._configs) == self._curr_config_index + 1
 
+    def _done_walking_max_batch_size(self):
+        if (self._max_batch_sizes is None or len(self._max_batch_sizes)
+                == self._curr_max_batch_size_index + 1):
+            return True
+
+        if self._early_exit_enable and not self._last_results_increased_throughput(
+        ):
+            if not self._max_batch_size_warning_printed:
+                logger.info(
+                    "No longer increasing max_batch_size because throughput has plateaued"
+                )
+                self._max_batch_size_warning_printed = True
+            return True
+        return False
+
     def _step(self):
+        if self._done_walking_max_batch_size():
+            self._reset_max_batch_size()
+            self._step_config()
+        else:
+            self._step_max_batch_size()
+
+    def _reset_max_batch_size(self):
+        self._curr_max_batch_size_index = 0
+        self._max_batch_size_warning_printed = False
+        self._curr_max_batch_size_throughputs = []
+
+    def _step_config(self):
         self._curr_config_index += 1
 
+    def _step_max_batch_size(self):
+        self._curr_max_batch_size_index += 1
+
+        last_max_throughput = self._get_last_results_max_throughput()
```

Comment:
Any reason last_max_throughput exists? Seems like these two lines can be combined

Reply:
just a bit easier to read. self._curr_max_batch_size_throughputs.append(self._get_last_results_max_throughput()) has a lot of redirects (and would span multiple lines anyways by the formatter)